File: 1408.string-matching-in-an-array.py
#
# @lc app=leetcode id=1408 lang=python3
#
# [1408] String Matching in an Array
#

# @lc code=start
class Solution:
    def stringMatching(self, words: List[str]) -> List[str]:
        return [s for s in words if any(True for i in words if i != s and s in i)]


# @lc code=end

File: cpp/1408.string-matching-in-an-array.cpp
/*
 * @lc app=leetcode id=1408 lang=cpp
 *
 * [1408] String Matching in an Array
 */
#include <vector>
#include <string>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<string> stringMatching(vector<string> &words)
    {
        int width = words.size();
        vector<string> res;
        for (auto &word : words)
        {
            for (int i = 0, word_len = word.size(); i != width; ++i)
            {
                if (words[i].size() >= word_len && words[i] != word && words[i].find(word) != -1)
                {
                    res.push_back(word);
                    break;
                }
            }
        }
        return res;
    }
};
// @lc code=end
